src/nvim/channel.c

```diff
@@ -563,6 +603,10 @@ size_t channel_send(uint64_t id, char *data, size_t len, bool data_owned, const
     goto retfree;
   }
 
+  if (chan->term) {
+    len += newline_adjust(&data);
```

Comment:
instead of doing this here, can we find where `data` is prepared and change its logic? similar to the simple change from https://github.com/neovim/neovim/pull/18723

Reply:
> So we could send a flag to `save_tv_as_string` that changes its separator char from `\n` to `\r\n` ?

That's what I did at first (see updates 4 and 5 in this [comment][1]). But `save_tv_as_string` handles putting text in general, so converting `\n` to `\r\n` will convert them when putting text in non-terminal buffers, too, which is undesired. We _could_ send a flag to determine if it's for a term channel, but: 1) I thought changing parameters of an existing function was too much of a hassle for a newbie in the codebase, and 2) I couldn't figure out if I should use `id`[2] or something else to figure out `chan->term` [3]. It's kind of about code design. I didn't wanna make big changes by shifting code around. Need guidance.

> `nvim_chan_send`, which only accepts String. I don't think we should be modifying raw string input--it's up to the caller to decide what raw data they want to send, right?

I actually didn't even see this. If we handle `\n` -> `\r\n` in `save_tv_as_string`, then this would be automatically fixed.

---

Sorry about adding links weirdly. GH is processing my links, prepending other links and stuff. Donno what that's about. Thought it best to paste it as text.


[1]: https://github.com/neovim/neovim/issues/18501#issuecomment-1174106059
2: https://github.com/3N4N/neovim/blob/85754dcdd992/src/nvim/eval/funcs.c#L873
3: https://github.com/neovim/neovim/blob/85754dcdd992/src/nvim/channel.c#L586

Reply:
@justinmk I pushed a second commit (we can squash it later; I thought we should keep the changes available until merged) following your suggestion: only converting LF to CRLF when joining list items.